Crash on Settings screens orientation change.

Here is the crash log when I change the orientation of the setting screens of all the built-in extensions. 

The reason is that on orientation change in `Activity`'s `onCreate` method, it recreates the `Fragment`. Which calls `setupSimplePreferencesScreen` on the `Activity`. When that is called, the line which initialize the `mFragment` instance is not called yet. 

```
E/AndroidRuntime(28106): FATAL EXCEPTION: main
E/AndroidRuntime(28106): Process: net.nurik.roman.dashclock, PID: 28106
E/AndroidRuntime(28106): java.lang.RuntimeException: Unable to start activity ComponentInfo{net.nurik.roman.dashclock/com.google.android.apps.dashclock.calendar.CalendarSettingsActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.google.android.apps.dashclock.configuration.BaseSettingsActivity$ExtensionPreferenceFragment.addPreferencesFromResource(int)' on a null object reference
E/AndroidRuntime(28106): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2325)
E/AndroidRuntime(28106): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2390)
E/AndroidRuntime(28106): 	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:3950)
E/AndroidRuntime(28106): 	at android.app.ActivityThread.access$900(ActivityThread.java:151)
E/AndroidRuntime(28106): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1309)
E/AndroidRuntime(28106): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(28106): 	at android.os.Looper.loop(Looper.java:135)
E/AndroidRuntime(28106): 	at android.app.ActivityThread.main(ActivityThread.java:5257)
E/AndroidRuntime(28106): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(28106): 	at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime(28106): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
E/AndroidRuntime(28106): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
E/AndroidRuntime(28106): Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.google.android.apps.dashclock.configuration.BaseSettingsActivity$ExtensionPreferenceFragment.addPreferencesFromResource(int)' on a null object reference
E/AndroidRuntime(28106): 	at com.google.android.apps.dashclock.calendar.CalendarSettingsActivity.setupSimplePreferencesScreen(CalendarSettingsActivity.java:37)
E/AndroidRuntime(28106): 	at com.google.android.apps.dashclock.configuration.BaseSettingsActivity$ExtensionPreferenceFragment.onCreate(BaseSettingsActivity.java:184)
E/AndroidRuntime(28106): 	at android.app.Fragment.performCreate(Fragment.java:2031)
E/AndroidRuntime(28106): 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:863)
E/AndroidRuntime(28106): 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1067)
E/AndroidRuntime(28106): 	at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1049)
E/AndroidRuntime(28106): 	at android.app.FragmentManagerImpl.dispatchCreate(FragmentManager.java:1864)
E/AndroidRuntime(28106): 	at android.app.Activity.onCreate(Activity.java:936)
E/AndroidRuntime(28106): 	at android.support.v4.app.FragmentActivity.onCreate(FragmentActivity.java:257)
E/AndroidRuntime(28106): 	at android.support.v7.app.AppCompatActivity.onCreate(AppCompatActivity.java:58)
E/AndroidRuntime(28106): 	at com.google.android.apps.dashclock.configuration.BaseSettingsActivity.onCreate(BaseSettingsActivity.java:49)
E/AndroidRuntime(28106): 	at android.app.Activity.performCreate(Activity.java:5990)
E/AndroidRuntime(28106): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1106)
E/AndroidRuntime(28106): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2278)
E/AndroidRuntime(28106): 	... 11 more
```

--- a/main/src/main/java/com/google/android/apps/dashclock/configuration/BaseSettingsActivity.java
+++ b/main/src/main/java/com/google/android/apps/dashclock/configuration/BaseSettingsActivity.java
@@ -65,6 +65,11 @@ public abstract class BaseSettingsActivity extends AppCompatActivity {
     }
 
     @Override
+    public void onAttachFragment(android.app.Fragment fragment) {
+        mFragment = (ExtensionPreferenceFragment) fragment;
+    }
+
+    @Override
     public boolean onOptionsItemSelected(MenuItem item) {
         if (item.getItemId() == android.R.id.home) {
             // TODO: if the previous activity on the stack isn't a ConfigurationActivity,

